src/app/_components/RolesList/RolesList.tsx

```diff
@@ -1,43 +1,45 @@
 'use client';
 
-import {useMemo} from "react";
-import {Roles} from "@/roles";
-import {Modifiers} from "@/modifiers";
-import {RoleOrModifierTypes} from "@/constants/rolesAndModifiers";
+import {type FC, useMemo} from "react";
+import {type Modifier, type Role} from "@/constants/rolesAndModifiers";
 import {Teams} from "@/constants/teams";
 import {RolesListContext, TeamFilters, TypeFilters, useFilters, useSearch} from "./hooks";
 import {Search} from "./Search";
 import {Filters} from "./Filters";
 import {RoleCard} from "./RoleCard/RoleCard";
 
-const RolesAndModifiers = [...Roles, ...Modifiers];
-
-export const RolesList = ({ hideElement = false }: { hideElement?: boolean }) => {
+export const RolesList: FC<{
+    roles: Role[],
+    modifiers: Modifier[],
```

Comment:
Generalnie jak kopiujemy cały kod (np. `RolesListTajemniczy`) to od razu znak, że coś jest do uwspólnienia. W takiej sytuacji najpierw szukamy części wspólnej kodu (w tym przypadku praktycznie cały kod) i części rozłącznej (sama lista ról i modyfikatorów, czyli źródło danych).

Teraz kod jest wspólny i nie trzeba robić zmian w dwóch lub więcej miejsc.